.. contents::
   :depth: 3.0
..

Going forward
=============

Look into
---------

-  `Practical Computing for Biologists book by Haddock and
   Dunn <http://practicalcomputing.org/>`__

-  `April 2013 Software Carpentry at
   Arizona <http://2013-swc-az.readthedocs.org/en/latest/index.html>`__

-  `Bioinf-py <http://hplgit.github.io/bioinf-py/doc/pub/html/index.html>`__
   At the `main site you can select your
   form <http://hplgit.github.io/bioinf-py/doc/web/index.html>`__.

-  `Interactive notebooks: Sharing the code by Helen Shen. Nature. 2014
   Nov 6;515(7525):151-2. doi: 10.1038/515151a. PMID:
   25373681 <http://www.nature.com/news/interactive-notebooks-sharing-the-code-1.16261>`__

-  `Programming tools: Adventures with R by Sylvia Tippmann. Nature.
   2015 Jan 1;517(7532):109-10. doi: 10.1038/517109a. PMID:
   25557714 <http://www.nature.com/news/programming-tools-adventures-with-r-1.16609>`__

-  `a two year-old screencast intro of IPython notebook by Titus
   Brown <https://www.youtube.com/watch?v=HaS4NXxL5Qc&feature=youtu.be>`__
   (Skip to the three-minute mark since we aren't necessarily interested
   in running it on Amazon web services right now.) A non-interactive
   version of the notebook he demonstrates is
   `here <http://nbviewer.ipython.org/github/fomightez/jan2015feng_gr_m/blob/master/others_demos/titus_screencast.ipynb>`__.

-  `Another take on the wonders of the IPython
   notebook <http://nbviewer.ipython.org/url/inesdesantiago.github.io/SeqQC.blog/ipythonSlides_post5/Example_Ipython.ipynb>`__,
   from `a
   blog <https://seqqc.wordpress.com/2015/02/22/make-slides-with-ipython-notebook/>`__.

-  `Orchestrating high-throughput genomic analysis with Bioconductor.
   Huber W, Carey VJ, Gentleman R, Anders S, Carlson M, Carvalho BS,
   Bravo HC, Davis S, Gatto L, Girke T, Gottardo R, Hahne F, Hansen KD,
   Irizarry RA, Lawrence M, Love MI, MacDonald J, Obenchain V, Oleś AK,
   Pagès H, Reyes A, Shannon P, Smyth GK, Tenenbaum D, Waldron L, Morgan
   M. Nat Methods. 2015 Jan 29;12(2):115-21. doi: 10.1038/nmeth.3252.
   PMID:
   25633503 <http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html>`__

-  `Programming: Pick up Python by Jeffrey M. Perkel. Nature. 2015
   February 5;518:125–6.doi:10.1038/518125a.
   PMID:25653001 <http://www.nature.com/news/programming-pick-up-python-1.16833>`__
   too

-  `A History of Bioinformatics (in the Year 2039), a presentation by
   Titus Brown encouraging good data practices in his unique
   way <https://www.youtube.com/watch?v=uwsjwMO-TEA>`__

Regular Expressions
~~~~~~~~~~~~~~~~~~~

-  `Chapters 2 & 3 of Practical Computing for Biologists book by Haddock
   and Dunn <http://practicalcomputing.org/>`__. The related appendix #2
   is freely available as part of `tables of Appendices from Practical
   Computing for Biologists book by Haddock and
   Dunn <http://practicalcomputing.org/files/PCfB_Appendices.pdf>`__

-  `Regular Expressions
   Primer <http://www.ternent.com/tech/regexp.html>`__

-  `Regular Expressions 101: online regex editor and debugger
   tool <https://regex101.com/>`__ (Seems best with ``g`` global
   modifier on.)

-  `RegExr v2.0: online tool to learn, build, & test Regular
   Expressions <http://regexr.com/>`__

-  `regex tester <http://rextester.com/tester>`__

-  `Python Regular Expression Testing
   Tool <http://www.pythonregex.com/>`__

-  For using regular expressions in Sublime Text, you need to click on
   the box ``.*`` next to ``Find What`` when Find tool open to activate.
   See
   `here <http://sublime-text-unofficial-documentation.readthedocs.org/en/latest/search_and_replace/search_and_replace_overview.html>`__
   for more information.

-  In TextWrangler the trick to activate Regular Expressions is to
   toggle on ``Grep`` box under ``Matching`` in the ``Find and Replace``
   panel. See `here about 1 and half minutes into the
   video <https://www.youtube.com/watch?v=vuDoJrpLnBk>`__.

-  `BBEdit-TextWrangler\_RegEx\_Cheat\_Sheet.txt <https://gist.github.com/ccstone/5385334>`__

-  See `here under 'Search and replace with special characters (regular
   expressions)' <http://oracc.museum.upenn.edu/doc/help/usingemacs/aquamacs/index.html>`__
   for using Regular Expressions in AquaMacs.

IPython Notebook
~~~~~~~~~~~~~~~~

-  `Interactive notebooks: Sharing the code by Helen Shen. Nature. 2014
   Nov 6;515(7525):151-2. doi: 10.1038/515151a. PMID:
   25373681 <http://www.nature.com/news/interactive-notebooks-sharing-the-code-1.16261>`__

-  `a two year-old screencast intro of IPython notebook by Titus
   Brown <https://www.youtube.com/watch?v=HaS4NXxL5Qc&feature=youtu.be>`__
   (Skip to the three-minute mark since we aren't necessarily interested
   in running it on Amazon web services right now.) A non-interactive
   version of the notebook he demonstrates is
   `here <http://nbviewer.ipython.org/github/fomightez/jan2015feng_gr_m/blob/master/others_demos/titus_screencast.ipynb>`__.

-  `Another take on the wonders of the IPython
   notebook <http://nbviewer.ipython.org/url/inesdesantiago.github.io/SeqQC.blog/ipythonSlides_post5/Example_Ipython.ipynb>`__,
   from `a
   blog <https://seqqc.wordpress.com/2015/02/22/make-slides-with-ipython-notebook/>`__
-  `The future of the IPython Notebook is the Jupyter
   project <http://jeroenjanssens.com/2015/02/19/ibash-notebook.html>`__

-  `Analyzing data with R in the IPython
   notebook <http://nbviewer.ipython.org/github/dboyliao/cookbook-code/blob/master/notebooks/chapter07_stats/08_r.ipynb>`__

R and Bioconductor in general
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

-  `Programming tools: Adventures with R by Sylvia Tippmann. Nature.
   2015 Jan 1;517(7532):109-10. doi: 10.1038/517109a. PMID:
   25557714 <http://www.nature.com/news/programming-tools-adventures-with-r-1.16609>`__

-  `Orchestrating high-throughput genomic analysis with Bioconductor.
   Huber W, Carey VJ, Gentleman R, Anders S, Carlson M, Carvalho BS,
   Bravo HC, Davis S, Gatto L, Girke T, Gottardo R, Hahne F, Hansen KD,
   Irizarry RA, Lawrence M, Love MI, MacDonald J, Obenchain V, Oleś AK,
   Pagès H, Reyes A, Shannon P, Smyth GK, Tenenbaum D, Waldron L, Morgan
   M. Nat Methods. 2015 Jan 29;12(2):115-21. doi: 10.1038/nmeth.3252.
   PMID:
   25633503 <http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html>`__

Learning R
~~~~~~~~~~

-  See the box in the article `Programming tools: Adventures with R by
   Sylvia Tippmann. Nature. 2015 Jan 1;517(7532):109-10. doi:
   10.1038/517109a. PMID:
   25557714 <http://www.nature.com/news/programming-tools-adventures-with-r-1.16609>`__.

-  The Coursera courses in `Johns Hopkins' Data Science
   Specialization <https://www.coursera.org/specialization/jhudatascience/1?utm_medium=courseDescripTop>`__,
   in particular the `R
   Programming <https://www.coursera.org/course/rprog>`__ and `Getting
   and Cleaning Data <https://www.coursera.org/course/getdata>`__
   courses. If you are brand new to this and don't yet know how to use
   Github, `The Data Scientist's
   Toolbox <https://www.coursera.org/course/datascitoolbox>`__ would
   probably be helpful as a starting point.

-  You do get a bit of flavor for the use of R in data analysis in the
   Coursera courses `Bioinformatic Methods
   I <https://www.coursera.org/course/bioinfomethods1>`__ and
   `Bioinformatic Methods
   II <https://www.coursera.org/course/bioinfomethods2>`__

-  `How to Transition from Excel to R: An Intro to R for Microsoft Excel
   Users <http://districtdatalabs.silvrback.com/intro-to-r-for-microsoft-excel-users>`__

ChIP-seq data analysis
~~~~~~~~~~~~~~~~~~~~~~

-  `Titus Brown and Colleague's Next-Gen Sequence Analysis
   Workshops <http://ged.msu.edu/angus/>`__, most recent is [Next-Gen
   Sequence Analysis Workshop
   (2014)(http://angus.readthedocs.org/en/2014/) Particularly pertinent
   are the sections `Day 7: ChIP-seq: Peak Predictions and
   Cis-regulatory Element
   Annotations <http://ged.msu.edu/angus/tutorials-2011/day7.html>`__,
   `Using MEME to identify TF binding motif from ChIP-seq
   data <http://ged.msu.edu/angus/tutorials/chipseq-motif-finding.html>`__
   and `here <http://ged.msu.edu/angus/tutorials-2012/day5.html>`__.

-  `ChIP- and DNase-seq data analysis workshop
   2014 <http://web.csc.fi/english/csc/courses/archive/chipseq2014>`__

-  `Cis-regulatory Element Annotation
   System <http://liulab.dfci.harvard.edu/CEAS/>`__ by Hyunjin Shin and
   Tao Liu from Xiaole Shirley Liu's Lab

-  `ab initio motif finder
   MEME <http://www.ncbi.nlm.nih.gov/pubmed/16845028>`__ and the related
   `MEME suite <http://www.ncbi.nlm.nih.gov/pubmed/19458158>`__

-  `MEME-LaB wraps the popular ab initio motif finder in a web
   tool <http://www.ncbi.nlm.nih.gov/pubmed/23681125>`__

-  `Motif enrichment tool. Blatti C, Sinha S. Nucleic Acids Res. 2014
   Jul;42(Web Server issue):W20-5. doi: 10.1093/nar/gku456. Epub 2014
   May 23. PMID:
   24860165 <http://www.ncbi.nlm.nih.gov/pubmed/24860165>`__

-  `Motif-based analysis of large nucleotide data sets using
   MEME-ChIP <http://www.ncbi.nlm.nih.gov/pubmed/24853928>`__

Plus see the 'literature' page in this collection of pages from the
session.

Python and ChIP-Seq and related analyses
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

-  `Feng et al. 2012. Identifying ChIP-seq enrichment using MACS.Nat
   Protoc. 2012 Sep; 7(9):
   10.1038/nprot.2012.101. <http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3868217/>`__

-  `Metaseq <https://github.com/daler/metaseq>`__ is a Python modle and
   covers `Determining Average ChIP-seq signal over promoters with
   Metaseq <http://nbviewer.ipython.org/github/daler/metaseq/blob/v0.5dev/doc/source/example_session.ipynb>`__.

-  NUCwave as described in `Comparative analysis of methods for
   genome-wide nucleosome cartography. Luis Quintales, Enrique Vázquez
   and Francisco Antequera. Briefings in Bioinformatics 2014. doi:
   10.1093/bib/bbu037 <http://nucleosome.usal.es/nucwave/>`__

-  `pyDNase - a library for analyzing DNase-seq
   data <http://pythonhosted.org/pyDNase/>`__ as described in Piper et
   al. 2013. Wellington: A novel method for the accurate identification
   of digital genomic footprints from DNase-seq data, Nucleic Acids
   Research 2013; doi: 10.1093/nar/gkt850

    Many people currently analyzing DNase-seq data are using tools
    designed for ChIP-seq work, but may be inappropriate for DNase-seq
    data where one is less interested in the overlaps of sequenced
    fragments, but the site at which the cut occurs (the 5’ most end of
    the aligned sequence fragment). We have developed pyDNase to
    interface with a sorted and indexed BAM file from a DNase-seq
    experiment, allowing efficient and easy random access of DNase-seq
    cut data from any genomic location, e.g. Piper et al. 2013.
    Wellington: A novel method for the accurate identification of
    digital genomic footprints from DNase-seq data, Nucleic Acids
    Research 2013; doi: 10.1093/nar/gkt850

R and ChIP-seq
~~~~~~~~~~~~~~

-  `ChIPpeakAnno: a Bioconductor package to annotate ChIP-seq and
   ChIP-chip data. Lihua J Zhu12\*, Claude Gazin3, Nathan D Lawson12,
   Hervé Pagès4, Simon M Lin5, David S Lapointe6 and Michael R
   Green. <http://www.biomedcentral.com/1471-2105/11/237>`__

Git and Github
~~~~~~~~~~~~~~

-  `Github bootcamp <https://help.github.com/categories/bootcamp/>`__

-  `main Git resource <http://git-scm.com/>`__. You can even `learn git
   right in your
   browser <https://try.github.io/levels/1/challenges/1>`__

-  `Simple guide to git <http://rogerdudler.github.io/git-guide/>`__

-  `About Gists <https://help.github.com/articles/about-gists/>`__

Questions
~~~~~~~~~

-  Try Google, probably will lead you to one of my listed resources
   or...
-  `Biostars <https://www.biostars.org>`__
-  Stackoverflow for general scripting and computing
-  `SEQanswers <http://seqanswers.com/>`__ - a high throughput
   sequencing community

-  Try Twitter - for example
   `this <https://twitter.com/jaredtsimpson/status/535443211452702720>`__
